src/graph/visitor/DeduceTypeVisitor.cpp

```diff
@@ -94,6 +94,16 @@ static const std::unordered_map<Value::Type, Value> kConstantValues = {
   }                                                                  \
   type_ = detectVal.type()
 
+#define CHECK_DEPTH(depth)                                 \
```

Comment:
Better to replace by method.

Reply:
Sorry I am new in c++,  I wanna return in the macro and don't know how to replace it with a method

Reply:
As bellow:
```c++
checkDepth();
if (status_.ok() == false) {
    return;
}
```

Reply:
thanks！